sqldatabaseproductserver/main.go

```diff
@@ -0,0 +1,213 @@
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	for result.Next() {
+		var product Product
+		err := result.Scan(&product.ID, &product.Name, &product.Price)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		products = append(products, product)
+	}
+
+	err = json.NewEncoder(w).Encode(products)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func GetProduct(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	stringid := vars["id"]
+	intid, err := strconv.Atoi(stringid)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	if intid > 100 {
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+	stmt, err := db.Prepare("SELECT * FROM products WHERE ID =?;")
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	result := stmt.QueryRow(intid)
+	err = stmt.Close()
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	var product Product
+	err = result.Scan(&product.ID, &product.Name, &product.Price)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	response, err := json.Marshal(product)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	_, err = w.Write(response)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+}
+func AddProduct(w http.ResponseWriter, r *http.Request) {
+	request, err := io.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	newproduct := Product{
+		ID:    0,
+		Name:  "",
+		Price: 0,
+	}
+
+	err = json.Unmarshal(request, &newproduct)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	stmt, err := db.Prepare("INSERT INTO products (ID, Name, Price) VALUES (?,?,?);")
+	_, err = stmt.Exec(newproduct.ID, newproduct.Name, newproduct.Price)
+	w.WriteHeader(http.StatusCreated)
+
+}
+func UpdateProduct(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	stringid := vars["id"]
+	id, err := strconv.Atoi(stringid)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	if id > 100 {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	request, err := io.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	newproduct := Product{
+		ID:    0,
+		Name:  "",
+		Price: 0,
+	}
+
+	err = json.Unmarshal(request, &newproduct)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	stmt, err := db.Prepare("UPDATE products SET Name =?, Price =?, WHERE ID =?")
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	_, err = stmt.Exec(newproduct.Name, newproduct.Price, newproduct.ID)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
```

Comment:
No need for this. By default a `200 OK` is returned unless you specify otherwise.